packages/fx-core/src/core/middleware/utils/v3MigrationUtils.ts

```diff
@@ -278,7 +278,7 @@ export async function readAndConvertUserdata(
 ): Promise<string> {
   let returnAnswer = "";
 
-  const userdataContent = fs.readFileSync(path.join(context.projectPath, filePath), "utf8");
+  const userdataContent = await fs.readFile(path.join(context.projectPath, filePath), "utf8");
```

Comment:
The same question as Chaoyi's: why we need to make this change, they look the same

Reply:
It is ok to update this function to sync function. 
If so, we need to handle the thrown out error of readFile which the caller of the function should update as well.
If we use await inside the function, the caller of the function has no need to change.